src/eccentricity.js

```diff
@@ -0,0 +1,70 @@
+module.exports = eccentricity;
+
+/**
+ * The eccentricity centrality of a node is the greatest distance between that node and
+ * any other node in the network. 
+ */
+function eccentricity(graph, oriented) {
+  var Q = [];
+  // distance from source
+  var dist = Object.create(null);
+
+  var currentNode;
+  var centrality = Object.create(null);
+
+  graph.forEachNode(setCentralityToZero);
+  graph.forEachNode(calculateCentrality);
+
+  return centrality;
+
+  function setCentralityToZero(node) {
+    centrality[node.id] = 0;
+  }
+
+  function calculateCentrality(node) {
+    currentNode = node.id;
+    singleSourceShortestPath(currentNode);
+    accumulate();
+  }
+
+  function accumulate() {
+    // Add all distances for node to array, excluding -1s
+    var distances = Object.keys(dist).map(function(key) {return dist[key]}).filter(function(val){return val !== -1});

```

Comment:
Great way to simplify it into one loop indeed. One consequence of the above is that isolated nodes will have their ``maxDist`` set to -1 instead of 0. Since we never deal with negative distances, what about setting ``maxDist`` to 0 initially?